From 160ec2ca57d52ce55de1954ea0ff966e07774160 Mon Sep 17 00:00:00 2001
From: Srinivas Pandruvada <srinivas.pandruvada@linux.intel.com>
Date: Fri, 13 Jun 2014 13:13:53 -0700
Subject: iio: accel: kxcjk1013: Fix missing comparison operator

Since comparison operator is missing, for an invalid frequency
it will overun the buffer.

Signed-off-by: Srinivas Pandruvada <srinivas.pandruvada@linux.intel.com>
---
 drivers/iio/accel/kxcjk-1013.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/iio/accel/kxcjk-1013.c b/drivers/iio/accel/kxcjk-1013.c
index 879ceda..fe75220 100644
--- a/drivers/iio/accel/kxcjk-1013.c
+++ b/drivers/iio/accel/kxcjk-1013.c
@@ -267,7 +267,7 @@ static int kxcjk1013_convert_freq_to_bit(int val, int val2)
 {
 	int i;
 
-	for (i = 0; ARRAY_SIZE(samp_freq_table); ++i) {
+	for (i = 0; i < ARRAY_SIZE(samp_freq_table); ++i) {
 		if (samp_freq_table[i].val == val &&
 			samp_freq_table[i].val2 == val2) {
 			return samp_freq_table[i].odr_bits;
-- 
1.7.9.5

